server/render/index.js

```diff
@@ -152,7 +152,10 @@ export function serverRender( req, res ) {
 			stateCache.set( cacheKey, serverState );
 		}
 
-		context.lang = getCurrentLocaleSlug( context.store.getState() ) || context.lang;
+		context.lang =
+			getCurrentLocaleVariant( context.store.getState() ) ||
```

Comment:
I think I missed this last time, but, unless we're using the locale variant explicitly, I'm not convinced we need to set `context.lang` to the variant at all here.

`context.lang` is assigned and used in a few places, one example is setting the document title. With user bootstrapping on, this is the side effect of the above expression:

<img width="553" alt="screen shot 2018-03-13 at 12 45 33 pm" src="https://user-images.githubusercontent.com/6458278/37318828-a5012c70-26c0-11e8-8207-2748e8e6a96e.png">

Ideally it should be `de`.

I've tested with just the `localeSlug`, and there appears to be no side effects related to removing the locale variant check:

```
		context.lang = getCurrentLocaleSlug( context.store.getState() ) || context.lang;
```

There is one other issue however, and that is the value we assign to `context.i18nLocaleScript`. As far as I can tell, for user-bootstrapped pages we use the values in this script to [populate the locale strings](https://github.com/Automattic/wp-calypso/blob/151b0105a5734b6a60def82d5079370fa76afe28/client/boot/common.js#L144).

So, even with a locale variant set, the result is the following:

<img width="480" alt="screen shot 2018-03-13 at 12 59 55 pm" src="https://user-images.githubusercontent.com/6458278/37318835-b4736d8a-26c0-11e8-9cf1-e00faa66d56a.png">

I think we could get around this by [checking for the variant earlier at this line](https://github.com/Automattic/wp-calypso/blob/151b0105a5734b6a60def82d5079370fa76afe28/server/render/index.js#L116):

```
	if ( ! isDefaultLocale( context.lang ) ) {
		const langFileName = getCurrentLocaleVariant( context.store.getState() ) || context.lang;
		context.i18nLocaleScript = '//widgets.wp.com/languages/calypso/' + langFileName + '.js';
	}
```

Reply:
Nice catch! I've updated the code in f9a5904.